instrumentation/nocode/src/test/java/com/splunk/opentelemetry/instrumentation/nocode/JSPSTest.java

```diff
@@ -0,0 +1,205 @@
+    assertEquals("g value", JSPS.evaluate(jsps, o, new Object[0]));
+  }
+
+  public static class TakeString {
+    public String take(String s) {
+      return s;
+    }
+  }
+
+  public static class TakeObject {
+    public String take(Object o) {
+      return o.toString();
+    }
+  }
+
+  public static class TakeBooleanPrimitive {
+    public String take(boolean param) {
+      return Boolean.toString(param);
+    }
+  }
+
+  public static class TakeBoolean {
+    public String take(Boolean param) {
+      return param.toString();
+    }
+  }
+
+  public static class TakeIntegerPrimitive {
+    public String take(int param) {
+      return Integer.toString(param);
+    }
+  }
+
+  public static class TakeInteger {
+    public String take(Integer param) {
+      return param.toString();
+    }
+  }
+
+  public static class TakeLongPrimitize {
+    public String take(long param) {
+      return Long.toString(param);
+    }
+  }
+
+  public static class TakeLong {
+    public String take(Long param) {
+      return param.toString();
+    }
+  }
+
+  @Test
+  public void testBooleanLiteralParamTypes() {
+    TakeBooleanPrimitive b = new TakeBooleanPrimitive();
+    TakeBoolean B = new TakeBoolean();
+    TakeObject O = new TakeObject();
+    assertEquals("true", JSPS.evaluate("this.take(true)", b, new Object[0]));
+    assertEquals("false", JSPS.evaluate("this.take(false)", B, new Object[0]));
+    assertEquals("true", JSPS.evaluate("this.take(true)", O, new Object[0]));
+  }
+
+  @Test
+  public void testStringLiteralParamTypes() {
+    TakeString S = new TakeString();
+    TakeObject O = new TakeObject();
+    assertEquals("a", JSPS.evaluate("this.take(\"a\")", S, new Object[0]));
+    assertEquals("a", JSPS.evaluate("this.take(\"a\")", O, new Object[0]));
+  }
+
+  @Test
+  public void testIntegerLiteralParamTypes() {
+    TakeIntegerPrimitive i = new TakeIntegerPrimitive();
+    TakeInteger I = new TakeInteger();
+    TakeLongPrimitize l = new TakeLongPrimitize();
+    TakeLong L = new TakeLong();
+    TakeObject O = new TakeObject();
+    assertEquals("13", JSPS.evaluate("this.take(13)", i, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", I, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", l, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", L, new Object[0]));
+    assertEquals("13", JSPS.evaluate("this.take(13)", O, new Object[0]));
+  }
+
+  @Test
+  public void testWhitespace() {
+    String[] tests =
+        new String[] {
+          "this.get(\"key\").substring(1)",
+          " this.get(\"key\").substring(1)",
+          "this .get(\"key\").substring(1)",
+          "this. get(\"key\").substring(1)",
+          "this.get (\"key\").substring(1)",
+          "this.get( \"key\").substring(1)",
+          "this.get(\"key\" ).substring(1)",
+          "this.get(\"key\") .substring(1)",
+          "this.get(\"key\"). substring(1)",
```

Comment:
Makes me wonder if newlines and/or tabs would be allowed at this boundary. Might actually be a nice feature for longer blocks of code that is definitely not code. Maybe add test coverage for those?

Reply:
The implementation uses `Character.isWhitespace`; threw in a few other whitespace characters for good measure!